lib/services/datasets.js

```diff
@@ -0,0 +1,528 @@
+  if (options.description) invariant(typeof options.description === 'string', 'options.description must be a string');
+
+  var url = makeURL(this, constants.API_DATASET_DATASETS, { owner: this.user });
+
+  request
+    .post(url)
+    .send(options)
+    .end(function(err, res) {
+      callback(err, res.body);
+    });
+};
+
+/**
+ * To retrieve information about a particular dataset.
+ * This request requires an access token with the datasets:read scope.
+ *
+ * @param {string} dataset the id for an existing dataset
+ * @param {Function} callback called with (err, dataset)
+ * @returns {undefined} nothing, calls callback
+ * @example
+ * var mapboxClient = new MapboxClient('ACCESSTOKEN');
+ * mapboxClient.readDataset('dataset-id', function(err, dataset) {
+ *   console.log(dataset);
+ *   // {
+ *   //   "owner": {account},
+ *   //   "id": "dataset-id",
+ *   //   "name": {dataset name},
+ *   //   "description": {dataset description},
+ *   //   "created": {timestamp},
+ *   //   "modified": {timestamp}
+ *   // }
+ * });
+ */
+Datasets.prototype.readDataset = function(dataset, callback) {
+  invariant(typeof dataset === 'string', 'dataset must be a string');
+  invariant(typeof callback === 'function', 'callback must be a function');
+
+  var url = makeURL(this, constants.API_DATASET_DATASET, {
+    owner: this.user,
+    dataset: dataset
+  });
+
+  request(url, function(err, res) {
+    callback(err, res.body);
+  });
+};
+
+/**
+ * To make updates to a particular dataset's properties.
+ * This request requires an access token with the datasets:write scope.
+ *
+ * @param {string} dataset the id for an existing dataset
+ * @param {object} [options] an object defining updates to the dataset's properties
+ * @param {string} [options.name] the updated dataset's name
+ * @param {string} [options.description] the updated dataset's description
+ * @param {Function} callback called with (err, dataset)
+ * @returns {undefined} nothing, calls callback
+ * @example
+ * var mapboxClient = new MapboxClient('ACCESSTOKEN');
+ * var options = { name: 'foo' };
+ * mapboxClient.updateDataset('dataset-id', options, function(err, dataset) {
+ *   console.log(dataset);
+ *   // {
+ *   //   "owner": {account},
+ *   //   "id": "dataset-id",
+ *   //   "name": "foo",
+ *   //   "description": {dataset description},
+ *   //   "created": {timestamp},
+ *   //   "modified": {timestamp}
+ *   // }
+ * });
+ */
+Datasets.prototype.updateDataset = function(dataset, options, callback) {
+  invariant(typeof dataset === 'string', 'dataset must be a string');
+  invariant(typeof callback === 'function', 'callback must be a function');
+  invariant(typeof options === 'object', 'options must be an object');
+  invariant(!!options.name || !!options.description, 'options must include a name or a description');
+
+  if (options.name) invariant(typeof options.name === 'string', 'options.name must be a string');
```

Comment:
Should we do this type of validation here or let it happen on the server? I could see us making a backward compatible change that removes this requirement.